extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -160,8 +160,8 @@ export class PublishOptionsDialog {
 		const result = await this.publish.getDefaultDeploymentOptions();
 		this.optionsModel.deploymentOptions = result;
 
-		// This will update the Map table with default values
-		this.optionsModel.InitializeUpdateOptionsMapTable();
+		// This will update the boolean options map with default values
+		this.optionsModel.InitializeBooleanOptionsMap();
```

Comment:
Is there any time when you set deploymentOptions that you wouldn't want to call this immediately? If not then you should be turning deploymentOptions into a setter property and calling this there.